File: 백준/Silver/2606. 바이러스/바이러스.js
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";

const input = require("fs")
  .readFileSync(filePath)
  .toString()
  .trim()
  .split("\n");

const bfs = (start) => {
  let queue = [];
  queue.push(start);

  visited[start] = 1;

  while (queue.length) {
    let v = queue.shift();

    for (i of myGraph[v]) {
      if (!visited[i]) {
        queue.push(i);
        visited[i] = 1;
        count++;
      }
    }
  }
};

const N = +input.shift();
const M = +input.shift();

let count = 0;

const visited = Array(N + 1).fill(0);
let myGraph = [];

for (let i = 0; i < N + 1; i++) {
  myGraph.push(new Array());
}

for (let i = 0; i < M; i++) {
  let [x, y] = input[i].split(" ").map(Number);
  myGraph[x].push(y);
  myGraph[y].push(x);
}

bfs(1);

console.log(count);

File: 백준/Silver/2606. 바이러스/바이러스.py
N = int(input())
M = int(input())
graph = [[] for i in range(N+1)]
visited = [0] * (N+1)
answer = 0

for i in range(M):
    x,y = map(int,input().split())
    graph[x].append(y)
    graph[y].append(x)

def dfs(node):
    global answer
    visited[node] = 1
    answer = answer+1
    
    for i in graph[node]:
        if(visited[i] ==0) :
            dfs(i)

dfs(1)

print(answer - 1)